app/src/main/java/com/yesser_studios/death_switch/JsonDeathStorage.kt

```
@@ -0,0 +1,105 @@
+    val deathsDatabase: Flow<DeathsDatabase> = _deathsDatabase.asStateFlow()
+
+    suspend fun load() {
+        withContext(Dispatchers.IO) {
+            try {
+                if (file.exists()) {
+                    val content = file.readText()
+                    _deathsDatabase.value = json.decodeFromString<DeathsDatabase>(content)
+                } else {
+                    _deathsDatabase.value = DeathsDatabase()
+                }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)